### DISCUSSION TITLE
Do you run z/OS open tools on z/OS V2.4 or older?

### DISCUSSION THREAD

**Discussion by IgorTodorovskiIBM** [zopencommunity/meta#855] [Polls]
We're considering changing the min arch level to be z13 (-qARCH11 or -march=arch11). We are also planning to migrate to 2.5 as the min OS supported level.

For reference: (thanks @AnthonyGiorgio)
<img width="1177" alt="Screenshot 2024-08-29 at 10 55 33 AM" src="https://github.com/user-attachments/assets/2b4553a6-03a9-4608-98bc-7d713f312bfc">
https://www.ibm.com/support/pages/system/files/inline-files/IBM%20Mainframe%20Life%20Cycle%20History%20V2.14%20-%20October%2010%202023_0.pdf

**Comment by lbdyck** [comment]
I would vote for support for all current processors along with one previous generation.

**Comment by AnthonyGiorgio** [comment]
It would affect me at the moment as one of the LPARs in my test sysplex is at v2.4 on a z14.  When in the future this is rotated out and the next z/OS release is rotated in, I would have no problems with a 2.5 minimum.

**Comment by skurtz97** [comment]
I do, but I shouldn't, and it's in the process of being upgraded so it shouldn't matter by the time the deadline passes I hope.

I think a 2.5 target for the future makes a lot of sense given 2.4 goes out of support in (checking calendar....) about 10 days.

Given how much improvement there has been in the C libraries and whatnot post-2.4, it would seem to me that it would be a good thing to move along with making the base something like 2.5 and arch level 13. But I'm probably not the right person to speak on issues of maintainability or productivity or that sort of thing.

I suppose the point I'm trying to make is that postponing a drop of official support for 2.4 much past the end of September deadline doesn't seem to make all that much sense (that doesn't mean people should intentionally break things to teach me and Anthony a lesson 🤣....) at least for these open source versions of the tools.

Just my two cents 🤷.

**Comment by IgorTodorovskiIBM** [comment]
I would like to bring this discussion back to the forefront.

Any concerns if we move the min OS level from 2.4 to 2.5 and the min supported arch level from z12 to z13 for the zopen community tools?

Now that 2.4 is out of service, this might be a good opportunity to move up.

**Comment by IgorTodorovskiIBM** [comment]
Here's the list of additional 2.5 C LE functions:
```
int accept4
int asprintf
int clone
int dup3
int epoll_create
int epoll_create1
int epoll_ctl
int epoll_pwait
int epoll_wait
int eventfd
int fdclosedir
int flock
int fremovexattr
int fsetxattr
int fstatfs
int futimesat
int getentropy
int getopt_long
int getpwent_r
int inet_aton
int inotify_add_watch
int inotify_init
int inotify_init1
int inotify_rm_watch
int lremovexattr
int lsetxattr
int pipe2
int pivot_root
int prctl
int removexattr
int renameat2
int sethostname
int setns
int setxattr
int statfs
int syncfs
int umount2
int unshare
int vasprintf
long openat2
long syscall
ssize_t fgetxattr
ssize_t flistxattr
ssize_t getrandom
ssize_t getxattr
ssize_t lgetxattr
ssize_t listxattr
ssize_t llistxattr
int accept4
int asprintf
int clone
int dup3
int epoll_create
int epoll_create1
int epoll_ctl
int epoll_pwait
int epoll_wait
int eventfd
int fdclosedir
int flock
int fremovexattr
int fsetxattr
int fstatfs
int futimesat
int getentropy
int getopt_long
int getpwent_r
int inet_aton
int inotify_add_watch
int inotify_init
int inotify_init1
int inotify_rm_watch
int lremovexattr
int lsetxattr
int pipe2
int pivot_root
int prctl
int removexattr
int renameat2
int sethostname
int setns
int setxattr
int statfs
int syncfs
int umount2
int unshare
int vasprintf
long openat2
long syscall
ssize_t fgetxattr
ssize_t flistxattr
ssize_t getrandom
ssize_t getxattr
ssize_t lgetxattr
ssize_t listxattr
ssize_t llistxattr
```
